node/chainSimulator/chainSimulator.go

```diff
@@ -195,6 +201,41 @@ func (s *simulator) GetInitialWalletKeys() *dtos.InitialWalletKeys {
 	return s.initialWalletKeys
 }
 
+// AddValidatorKeys will add the provided validators private keys in the keys handler on all nodes
+func (s *simulator) AddValidatorKeys(validatorsPrivateKeys [][]byte) error {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+
+	for _, node := range s.nodes {
+		err := s.setValidatorKeysForNode(node, validatorsPrivateKeys)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (s *simulator) setValidatorKeysForNode(node process.NodeHandler, validatorsPrivateKeys [][]byte) error {
+	for idx, privateKey := range validatorsPrivateKeys {
+
+		err := node.GetCryptoComponents().ManagedPeersHolder().AddManagedPeer(privateKey)
+		if err != nil {
+			return fmt.Errorf("cannot add private key for shard=%d, index=%d, error=%s", node.GetShardCoordinator().SelfId(), idx, err.Error())
+		}
+	}
+
+	return nil
+}
+
+// GetValidatorPrivateKeys will return the initial validators private keys
+func (s *simulator) GetValidatorPrivateKeys() []crypto.PrivateKey {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
```

Comment:
In this case, I want to keep the normal Lock because something in the state of the nodes has changed, and I want to stop the execution for other requests that are coming in parallel.